src/bentoml_cli/env_manager.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+import re
+import sys
+import typing as t
+import logging
+import functools
+
+import click
+
+from bentoml.exceptions import NotFound
+
+from bentoml._internal.env_manager import EnvManager
+
+logger = logging.getLogger(__name__)
+
+
+def remove_env_arg(cmd_args: list[str]) -> list[str]:
+    """
+    Removes `--env <env_name>` or `--env=<env_name>` from sys.argv
+    """
+    indices_to_remove: list[int] = []
+    for i, arg in enumerate(cmd_args):
+        # regex matching click.option
+        if re.match(r"^--env=?.*", arg):
+            indices_to_remove.append(i)
+            if "=" not in arg:
+                indices_to_remove.append(i + 1)
+
+    new_cmd_args: list[str] = []
+    for i, arg in enumerate(cmd_args):
+        if i not in indices_to_remove:
+            new_cmd_args.append(arg)
+    return new_cmd_args
```

Comment:
Should this be handled by `click.option`?

Reply:
the env_manager has to clear this one up before passing the args to the subroutine.